### PR TITLE
fix spark-rbac

### PR DESCRIPTION
**Pull request by Aransh**
## Purpose of this PR
A recent PR has added support for multiple namespace.
Among the changes it has added the field roleRef.namespace to a roleBinding, this field does not exist and causes issues with deployments. It is also not needed as the roleBinding resource is namespaced to begin with.
You can see a similar discussion here - https://stackoverflow.com/questions/51961432/cant-to-add-namespace-field-to-roleref-in-rolebinding

**Proposed changes:**
- Remove roleRef.namespace from spark roleBinding

## Change Category
Indicate the type of change by marking the applicable boxes:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could affect existing functionality)
- [ ] Documentation update

## Checklist
Before submitting your PR, please review the following:

- [x] I have conducted a self-review of my own code.
- [ ] I have updated documentation accordingly.
- [x] I have added tests that prove my changes are effective or that my feature works.
- [ ] Existing unit tests pass locally with my changes.

### CODE REVIEW

**Comment by MatheusAfinovicz** [comment]
+1, had the same issue this morning.
Thanks @Aransh 👍🏻

File: charts/spark-operator-chart/templates/spark-rbac.yaml [hunk 1]
```diff
@@ -49,7 +49,6 @@ subjects:
 roleRef:
   kind: Role
   name: spark-role
-  namespace: {{ $jobNamespace }}
```
Comment (vara-bonthu) [inline charts/spark-operator-chart/templates/spark-rbac.yaml:52]: @ananth102 This was added as a part of the support of multiple namespace here https://github.com/kubeflow/spark-operator/commit/835281d4d3110d265687a9d5b7504a261538af10. Did you validate if this is working for you? Please see the PR raised by @Aransh

Comment (AndrewChubatiuk) [reply charts/spark-operator-chart/templates/spark-rbac.yaml:52]: it's not needed, thanks for removing it

**Comment by vara-bonthu** [comment]
@Aransh, could you resolve the conflicts so we can proceed with merging the PR? Thank you!

**Comment by Aransh** [comment]
@vara-bonthu bumped version for this and https://github.com/kubeflow/spark-operator/pull/1976

**Comment by vara-bonthu** [comment]
It does look like the check are still failing. Coudl you please run the following in your local machine

https://github.com/kubeflow/spark-operator/blob/master/docs/developer-guide.md#run-helm-chart-lint

and `pre-commit`  https://github.com/kubeflow/spark-operator/blob/master/docs/developer-guide.md#run-helm-chart-lint

We may need to document this better in PR templates

**Comment by Aransh** [comment]
@vara-bonthu yup seems I'm a bit lost with this 😵‍💫
Every time I bump the version it seems to just ask me to bump it again, anyway ran these tests locally hopefully it's fine now

**Review by vara-bonthu — APPROVED** [review]
/lgtm

**Pull request merged**